### PR TITLE
fix: [DEL-4183]: Change Delegate useragents

### PR DESCRIPTION
**Pull request by jennyjames21**
## Describe your changes
Summary:
Add user agent as header to web socket/streaming calls
## Checklist
- [x ] I've documented the changes in the PR description.
- [x ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35716)

### CODE REVIEW

**Comment by jennyjames21** [comment]
image

**Comment by jennyjames21** [comment]
trigger messagecheck

File: 260-delegate/src/main/java/io/harness/delegate/service/DelegateAgentServiceImpl.java [hunk 1]
```diff
@@ -758,7 +758,8 @@ private RequestBuilder prepareRequestBuilder() {
       URI uri = uriBuilder.build();
 
       // Stream the request body
-      final RequestBuilder requestBuilder = client.newRequestBuilder().method(METHOD.GET).uri(uri.toString());
+      final RequestBuilder requestBuilder =
+          client.newRequestBuilder().method(METHOD.GET).header("User-Agent", "delegate_agent").uri(uri.toString());
```
Comment (johannesHarness) [inline 260-delegate/src/main/java/io/harness/delegate/service/DelegateAgentServiceImpl.java:762]: should we use the same as we use for rest calls?
`"delegate/" + this.versionInfoManager.getVersionInfo().getVersion()` in `420-delegate-agent/src/main/java/io/harness/managerclient/DelegateAgentManagerClientFactory.java`

**Comment by jennyjames21** [comment]
image

**Comment by jennyjames21** [comment]
*[`260-delegate/src/main/java/io/harness/delegate/service/DelegateAgentServiceImpl.java` line 762 at r1](https://reviewable.io/reviews/harness/harness-core/35716#-N7sn6s_bQ6rxqGl2P6B-r1-762:-N80RDaSDCB1GSMqj4YB:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/baa54a6dfc87b51b64976427872f1541354094ed/260-delegate/src/main/java/io/harness/delegate/service/DelegateAgentServiceImpl.java#L762)):*
<details><summary><i>Previously, johannesHarness (Johannes Batzill) wrote…</i></summary><blockquote>

should we use the same as we use for rest calls?
`"delegate/" + this.versionInfoManager.getVersionInfo().getVersion()` in `420-delegate-agent/src/main/java/io/harness/managerclient/DelegateAgentManagerClientFactory.java`
</blockquote></details>

Done.

**Review by johannesHarness — COMMENTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35716)* status: 1. unresolved discussions. 0 of 1 approvals obtained.

**Review by johannesHarness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35716)* status: :shipit: complete! ✓

**Pull request merged**